.github/workflows/_deploy-hl7-vpn-tunnel-cdk.yml

```diff
@@ -0,0 +1,110 @@
+name: Deploy VPN tunnel for HL7 Notification CDK
+
+# This workflow checks out both the OSS and Internal repos.
+# Because of that, the each step has to indicate the folder in which it should run at - different
+# from regular workflows that only checkout the repo once and run all steps in the same folder.
+
+on:
+  workflow_call:
+    inputs:
+      deploy_env:
+        description: "The environment we're deploying to, either 'staging' or 'production', this deploy does not run in sandbox"
+        required: true
+        type: string
+      partner_name:
+        description: "The partner we're deploying the tunnel for, e.g. 'MyTestHIE'"
+        required: true
+        type: string
+    secrets:
+      SERVICE_PAT:
+        required: true
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      SENTRY_AUTH_TOKEN:
+        required: false
+      SENTRY_ORG:
+        required: false
+      DOCKERHUB_USERNAME:
+        required: true
+      DOCKERHUB_TOKEN:
+        required: true
+
+jobs:
+  deploy:
+    # prevents 2+ devs/workflows trying to deploy to AWS at the same time
+    # https://serverlessfirst.com/emails/how-to-prevent-concurrent-deployments-of-serverless-stacks-in-github-actions/
+    # TODO Consider the solution here: https://github.com/tj-actions/aws-cdk/blob/main/action.yml
+    concurrency:
+      group: ${{ format('{0}-{1}', inputs.partner_name, inputs.deploy_env) }}
+    runs-on: ubuntu-latest
+    env:
+      AWS_REGION: ${{ inputs.deploy_env == 'production' && vars.API_REGION_PRODUCTION || vars.API_REGION_STAGING }}
```

Comment:
Why not use the same approach as `_deploy-cdk.yml`, so we don't have this workflow knowing about existing, env-specific vars?

Reply:
great point.